studio/src/ui/popoverMenus/PopoverManager.js

```diff
@@ -119,17 +117,23 @@ export class PopoverManager {
 	}
 
 	#hasBodyClickListener = false;
+
 	/**
 	 * @param {MouseEvent} e
 	 */
 	#onBodyClick = e => {
-		if (this.current && !this.current.el.contains(/** @type {Node} */ (e.target))) {
-			this.closeCurrent();
+		if (this.#activePopovers.length === 0) {
+			throw new Error("Error handling body click: No popovers exist");
 		}
+
+		this.#activePopovers.forEach(p => {
```

Comment:
It seems like there's an issue with how closing multiple popovers works, where opening an EntryPointPopover and then another will instantly close that second popover.

It seems like the onBodyClick doesn't have a good enough "filter" for figuring out whether a mouse event should close popovers or perform some function (like opening a new one)

I had some trouble figuring out a good set of document tag filters because a lot of the tags (like a button for example) are just divs with classes. I think we might have to leave this for now and revisit it as an MVP goal.

Reply:
Ah yes that makes sense. I think this seems fine for now. Especially since there's only a single situation where you might need a popover to stay open while doing other things (the entry point popover).

On another note, you can now right click the entry point droppable gui to get access to its context menu.
![image](https://user-images.githubusercontent.com/2737650/230782258-efa0d2bf-0193-4a3c-992b-b67659fc2004.png)
And all error messages are gone now as well so overall these are some pretty great improvements!